src/pymovements/dataset/dataset_definition.py

```diff
@@ -204,47 +204,45 @@ def from_yaml(path: str | Path) -> DatasetDefinition:
                 eyetracker=eyetracker,
             )
 
-        def reverse_substitute_types(d: Any) -> Any:
-            if isinstance(d, dict):
-                return {k: reverse_substitute_types(v) for k, v in d.items()}
-            if isinstance(d, list):
-                return [reverse_substitute_types(v) for v in d]
-            if isinstance(d, str) and d.startswith('!'):
-                type_name = d[1:]
-                if '.' in type_name:
-                    module_name, class_name = type_name.rsplit('.', 1)
-                    module = importlib.import_module(module_name)
-                    return getattr(module, class_name)
-                return getattr(builtins, type_name)
-            return d
-
         data = reverse_substitute_types(data)
         # Initialize DatasetDefinition with YAML data
         return DatasetDefinition(**data)
 
-    def to_yaml(self, path: str | Path) -> None:
-        """Save a dataset definition to a YAML file.
+    def to_dict(self, hide_private: bool = True) -> dict[str, Any]:
+        """Return dictionary representation.
 
         Parameters
         ----------
-        path: str | Path
-            Path where to save the YAML file to.
+        hide_private: bool
```

Comment:
I'm unsure about the argument name.

Another option could be `omit_private`. I don't like `skip_private` that much.

Reply:
done